### PR TITLE
Setup

### LINKED ISSUE #2: Setup
### Templates
- [ ] add issue templates (feature / tracker / bug)
- [ ] add pr template (feature / tracker / bugfix) 

### Workflows
- [ ] setup build > test Workflow
- [ ] add status batch to readme
- [ ] setup build > pack > publish Workflow
- [ ] add version batch to readme

### LICENCE.md

- [ ] Create

### CONTRIBUTING.md

- [ ] Create
- [ ] Include Structure
- [ ] Include Naming Conventions
- [ ] Include Style Guidelines

### CODE_OF_CONDUCT.md

- [ ] Create
- [ ] Contact method

## Branches

- [ ] Protect `master` (maybe rename to `main`)
- [ ] add `staging`

## Wiki

- [ ] Create

...to be continued

### PR DESCRIPTION
**Pull request by tweakch**
## Description

 

I created a structure for the SDK and generated the `BlockfrostService` and models from OpenApi specs (`v0.1.25`)

Fixes #2 

### Type of change

 
- [X] init

## Tests

- [x] Tests the `/health` endpoint using the `BlockfrostService`

## Pre Merge Todos

- [ ] My code follows the style guidelines of this project (see #4) 
- [X] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tweakch** [comment]
@mmahut feel free to add to #2

Id rename `master` to `staging` pretty soon and create a new `main` branch from it once this is merged but you guys decide...

**Comment by tweakch** [comment]
We'd also have to settle on the naming conventions for namespaces.and models. 

Since I gemerated the Service the models still look pretty wild.  

I will follow the asp.net api-guidelines as good as i can but its a lot of work.

 I'm a bit worried about maintainability with my approach, so If you have better tooling to generate from the specs or want to update them to be 100% consistent and unambiguous then go ahead and regenerate.

Really the only thing that needs to stay consistent iacross minor updates s the `IBlockfrostService` and  it is now in `v0.1.0-alpha` ^^

Yeah... let me know what you and the team thinks

File: README.md [hunk 1]
```diff
@@ -1,2 +1,74 @@
...
+
+To use this SDK, you first need login into to [blockfrost.io](https://blockfrost.io) create your project to retrive your API token.
+
+<img src="https://i.imgur.com/smY12ro.png">
+
+<br/>
+
+## Installation
+
+The SDK is hosted on [nuget.com](https://www.nuget.com/package/@blockfrost/blockfrost-dotnet), so you can directly import it using your favorite package manager.
```
Comment (eddex) [inline README.md:27]: Should be nuget.org not nuget.com

Comment (tweakch) [reply README.md:27]: hopp schwiiz 👍

**Comment by soyfrien** [comment]
Hello @tweakch, I am just looking through how you managed to turn the same Nswag generated code that I had, into something to start working with, whereas I was pretty stuck.

It's really interesting to see what you have done and I am trying to study it so I don't get stuck so early on the next time I try to use the tool on a swagger.json file.

I am just curious, did you separate things out for readability/style, and did you have to actually change any of the generated code?

I have been scrolling through your commits and see that you put a lot of thought into planning this solution, but could you give me a hint for the next time I encounter an OpenAPI that I want to consume with Nswag?

Many thanks!

**Pull request merged**